letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
```

Comment:
I think the problem with infinite loop was because `self.dv_c.remove(achall)` and `self.cont_c.remove(achall)` is not called by `_cleanup_challenges`, if challenges corresponding to `None` and `False` are not added to `active_achalls` - `AuthHandler` will continue to operate until `self.dv_c` and `self.cont_c` are empty... Thoughts? How should it be fixed?